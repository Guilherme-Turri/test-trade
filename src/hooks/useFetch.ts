import React from 'react'

export const useFetchData = () => {
  const [error, setError] = React.useState<string | null>(null);
  const [loading, setLoading] = React.useState<boolean>(false);

  const request = React.useCallback(async <T> (url: string, options: RequestInit) => {
    let response;
    let json:T | null;
    setLoading(true)
    try {
      setError(null)
      response = await fetch(url, options)
      json = await response.json()
      if (!response.ok) throw new Error()
    } catch (e: any) {
      json = null;
      setError('Falha ao conectar com o servidor.')
    }
    finally {
      setLoading(false)
          }
    return { response, json } 
  }, []);
  return { error, request, loading }
} 