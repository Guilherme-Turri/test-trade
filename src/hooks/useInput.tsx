import React from 'react'

const typeField = {
  email: {
    regex: /^.{2,}$/,
    message: "Por favor, digite duas letras ou mais."
  }
}

export const useInput = () => {
  const [value, setValue] = React.useState<string>('')
  const [error, setError] = React.useState<string>('');

  const validate = (value: string) => {
    if (typeField.email.regex.test(value) === false) setError(typeField.email.message)
    else setError('');
  }

  const onChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
    setValue(target.value)
    if (target.placeholder === "Nome") {
      validate(target.value)
    }
  }

  return (
    { value, onChange, error, setValue }
  )
}