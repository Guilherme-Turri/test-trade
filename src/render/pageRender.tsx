import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};