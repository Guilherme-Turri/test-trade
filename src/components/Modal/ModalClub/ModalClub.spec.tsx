/* eslint-disable testing-library/no-unnecessary-act */
import { pageRender } from "../../../render/pageRender";
import { fireEvent, screen, waitFor, act } from '@testing-library/react'
import { ModalClub } from ".";
import * as reactRedux from 'react-redux'

import * as fetchDataTest from '../../../hooks/useFetch'

jest.setTimeout(10000)

const mockedUseDispatch = jest.fn()
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: () => mockedUseDispatch
}));

const user = {
  apiKey: '123'
}

const parameters = [
  {
    team: {
      id: '01',
      name: 'Club01',
      logo: 'logo/01',
      code: '01'
    },
  },
  {
    team: {
      id: '02',
      name: 'Club02',
      logo: 'logo/02',
      code: '02'
    },
  }
]

describe('</ModalClub>', () => {
  const useSelectorMock = reactRedux.useSelector as jest.Mock;
  it('should render on screen properly', () => {
    useSelectorMock.mockReturnValue({
      parameters: {
        club: {
          all: parameters
        }
      }
    });
    const { container } = pageRender(<ModalClub />)
    const item = screen.getByText('Club01');
    expect(item).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
  it('Should render on the screen with a message to redo the search.', () => {
    useSelectorMock.mockReturnValue({
      parameters: null
    });
    pageRender(<ModalClub />)
    const item = screen.getByText('Refaça a busca e/ou verifique as requisições diárias');
    expect(item).toBeInTheDocument();
  })
  it('should dispatch selected club into context', () => {
    useSelectorMock.mockReturnValue({
      user,
      parameters: {
        club: {
          all: parameters
        }
      }
    });

    pageRender(<ModalClub />)

    const selectedClub = screen.getByText('Club01');
    act(() => {
      fireEvent.click(selectedClub);
    });

    expect(mockedUseDispatch).toHaveBeenCalledWith({
      "payload": {
        "team": {
          "code": "01",
          "id": "01",
          "logo": "logo/01",
          "name": "Club01",
        }
      }, "type": "parameters/setClub"
    });
  })
})