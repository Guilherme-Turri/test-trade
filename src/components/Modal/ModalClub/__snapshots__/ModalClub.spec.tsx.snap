// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</ModalClub> should render on screen properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.938rem;
  overflow-y: auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 9.375rem;
  height: 9.375rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 12.5rem;
  height: 12.5rem;
  margin: 0.938rem;
  cursor: pointer;
  border: 1px solid #00c555;
  border-radius: 0.625rem;
  background-color: #F3F4F5;
}

.c3 {
  color: #2d5839;
}

<div>
  <div
    class="c0"
    data-testid="modalClubId"
  >
    <div
      class="c1"
    >
      <img
        class="c2"
        src="logo/01"
      />
      <p
        class="c3"
      >
        Club01
      </p>
    </div>
    <div
      class="c1"
    >
      <img
        class="c2"
        src="logo/02"
      />
      <p
        class="c3"
      >
        Club02
      </p>
    </div>
  </div>
</div>
`;
